lib/backend/k8s/k8s_client_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2016-2020 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var _ = Describe("CreateKubernetesClientset fillLoadingRulesFromKubeConfigSpec", func() {
+
+	When("There are multiple Kubeconfig files specified", func() {
```

Comment:
No need to change this because you've only got one test in each block and it would just add a lot of boilerplate(!) but once you've got more than one test in a When/Describe/Context block, it should look like this:
```
When/Describe/Context("there are multiple foos", func() {
  var foos []foo
  var theThing thing
  BeforeEach(func() {
    foos = makeTheFoos()
    theThing = thing.New(thing.WithFoos(foos))
  })
  It("should do bar", func() {
    Expect(theThing).To(DoBar())
  })
  It("should do biff", func() {
    Expect(theThing).To(DoBiff())
  })
})
```
i.e. where you have `When("xyz", func()...)`, "xyz" should be done by the `BeforeEach` block nested inside the `When` so that hte set-up is shared between all the `It`s

Reply:
Right, forgot about BeforeEach! I'll just change it, seems to be structurally better even if it has a lot of boilerplate